### PR TITLE
added_int_bitwise_operator

### PR DESCRIPTION
**Pull request by ramvikrams**
added int in bitwise operators for issue #348

- [ ] Closes #xxxx (Replace xxxx with the Github issue number)
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Comment by Dr-Irv** [comment]
You're going to have to find a way to have both the changes to `core/series.pyi` and `tests/test_series.py` in one PR.  Choose this one or the other one.

**Comment by Dr-Irv** [comment]
xfef #375

**Comment by ramvikrams** [comment]
added both the files sir in this pr

File: pandas-stubs/core/series.pyi [hunk 1]
```diff
@@ -1226,23 +1226,23 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
     def __mod__(self, other: num | _ListLike | Series[S1]) -> Series[S1]: ...
     def __ne__(self, other: object) -> Series[_bool]: ...  # type: ignore[override]
     def __pow__(self, other: num | _ListLike | Series[S1]) -> Series[S1]: ...
-    def __or__(self, other: _ListLike | Series[S1]) -> Series[_bool]: ...
+    def __or__(self, other: int | _ListLike | Series[S1]) -> Series[int]: ...
     def __radd__(self, other: num | _str | _ListLike | Series[S1]) -> Series[S1]: ...
-    def __rand__(self, other: num | _ListLike | Series[S1]) -> Series[_bool]: ...
+    def __rand__(self, other: int | num | _ListLike | Series[S1]) -> Series[int]: ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/series.pyi:1231]: can you remove `num` from `__rand__`

File: pandas-stubs/core/series.pyi [hunk 2]
```diff
@@ -1226,23 +1226,23 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
...
     def __rdiv__(self, other: num | _ListLike | Series[S1]) -> Series[S1]: ...
     def __rdivmod__(self, other: num | _ListLike | Series[S1]) -> Series[S1]: ...
     def __rfloordiv__(self, other: num | _ListLike | Series[S1]) -> Series[S1]: ...
     def __rmod__(self, other: num | _ListLike | Series[S1]) -> Series[S1]: ...
     def __rmul__(self, other: num | _ListLike | Series) -> Series: ...
     def __rnatmul__(self, other: num | _ListLike | Series[S1]) -> Series[S1]: ...
     def __rpow__(self, other: num | _ListLike | Series[S1]) -> Series[S1]: ...
-    def __ror__(self, other: num | _ListLike | Series[S1]) -> Series[_bool]: ...
+    def __ror__(self, other: int | num | _ListLike | Series[S1]) -> Series[int]: ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/series.pyi:1239]: can you remove `num` from `__ror__`

File: pandas-stubs/core/series.pyi [hunk 3]
```diff
@@ -1226,23 +1226,23 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
...
     def __rsub__(self, other: num | _ListLike | Series[S1]) -> Series: ...
     @overload
     def __rtruediv__(self, other: Timedelta | TimedeltaSeries) -> Series[float]: ...
     @overload
     def __rtruediv__(self, other: num | _ListLike | Series[S1]) -> Series: ...
-    def __rxor__(self, other: num | _ListLike | Series[S1]) -> Series[_bool]: ...
+    def __rxor__(self, other: int | num | _ListLike | Series[S1]) -> Series[int]: ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/series.pyi:1245]: can you remove `num` from `__rxor__`

File: tests/test_series.py [hunk 4]
```diff
@@ -1236,3 +1236,24 @@ def cond2(x: pd.Series[int]) -> pd.Series[bool]:
 
     cond3 = pd.Series([False, True, True])
     check(assert_type(s.where(cond3, other=0), "pd.Series[int]"), pd.Series, int)
+
+
+def test_bitwise_operators_int() -> None:
```
Comment (Dr-Irv) [inline tests/test_series.py:1241]: change name to `test_bitwise_operators()`

File: tests/test_series.py [hunk 5]
```diff
@@ -1236,3 +1236,24 @@ def cond2(x: pd.Series[int]) -> pd.Series[bool]:
...
+    s = pd.Series([1, 2, 3, 4])
+#     for issue #348 (bitwise operators on Series should support int)
```
Comment (Dr-Irv) [inline tests/test_series.py:1243]: add back the tests such as `s & 3`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: tests/test_series.py (hunk 5 above)
Comment (ramvikrams) [reply tests/test_series.py]: done

File: tests/test_series.py (hunk 4 above)
Comment (ramvikrams) [reply tests/test_series.py]: done

File: pandas-stubs/core/series.pyi (hunk 1 above)
Comment (ramvikrams) [reply pandas-stubs/core/series.pyi]: done

File: pandas-stubs/core/series.pyi (hunk 2 above)
Comment (ramvikrams) [reply pandas-stubs/core/series.pyi]: done

File: pandas-stubs/core/series.pyi (hunk 3 above)
Comment (ramvikrams) [reply pandas-stubs/core/series.pyi]: done

File: tests/test_series.py [hunk 6]
```diff
@@ -1238,9 +1238,11 @@ def cond2(x: pd.Series[int]) -> pd.Series[bool]:
     check(assert_type(s.where(cond3, other=0), "pd.Series[int]"), pd.Series, int)
 
 
-def test_bitwise_operators_int() -> None:
+def test_bitwise_operators() -> None:
     s = pd.Series([1, 2, 3, 4])
-#     for issue #348 (bitwise operators on Series should support int)
+    # for issue #348 (bitwise operators on Series should support int)
+    check(assret_type(s & 3, "pd.Series[int]"), pd.Series,int)
+    check(assert_type(3 & s, "pd.Series[int]"), pd.Series,int)
```
Comment (Dr-Irv) [inline tests/test_series.py:1245]: Please add the tests for the `|` and `^` operator as well, as you had in the other PR

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: tests/test_series.py (hunk 6 above)
Comment (ramvikrams) [reply tests/test_series.py:1245]: done sir

File: tests/test_series.py [hunk 7]
```diff
@@ -1236,3 +1236,32 @@ def cond2(x: pd.Series[int]) -> pd.Series[bool]:
 
     cond3 = pd.Series([False, True, True])
     check(assert_type(s.where(cond3, other=0), "pd.Series[int]"), pd.Series, int)
+
+
+def test_bitwise_operators() -> None:
+    s = pd.Series([1, 2, 3, 4])
+    # for issue #348 (bitwise operators on Series should support int)
+    check(assret_type(s & 3, "pd.Series[int]"), pd.Series,int)
```
Comment (Dr-Irv) [inline tests/test_series.py:1244]: you misspelled `assert_type()` in 3 places.

You should test locally with `poetry run poe test` before committing.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: tests/test_series.py (hunk 7 above)
Comment (ramvikrams) [reply tests/test_series.py]: Sorry sir small mistake on my local setup I wrote it correctly but after copying here I unknowingly changed the spelling

**Comment by Dr-Irv** [comment]
You need to run `poetry run poe style` on these last changes.

**Comment by ramvikrams** [comment]
done sir

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
In `test_series.py`, you either have an extra blank or line at the end of the file, or missing a blank line.  I can't tell.

Please run `poetry run poe style`, make a commit and push.

**Comment by ramvikrams** [comment]
done sir

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Now you have extra spaces on the 2 lines in `tests/test_series.py` that precede `def test_bitwise_operators() -> None`

**Comment by ramvikrams** [comment]
> Now you have extra spaces on the 2 lines in `tests/test_series.py` that precede `def test_bitwise_operators() -> None`

I also don't know what is happening sir when I run `poetry run poe style` on my local system it shows no error

**Comment by ramvikrams** [comment]
Sir ideally two lines gap should be there?

**Comment by Dr-Irv** [comment]
> Sir ideally two lines gap should be there?

The gap should be there, but somehow you have blank spaces on the lines.

Do a `poetry update -vvv` to make sure you have latest versions of `black`, etc., installed, then `poetry run poe style`, then commit and push.

**Comment by ramvikrams** [comment]
sir done everything hope it works

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Still failing.  If you look here, you will see a spurious character at the beginning of line 1241
https://github.com/pandas-dev/pandas-stubs/pull/378/files/47a808f90e28eb0f8ced7f8f79bd93c624ed75ac..b013b7a9fabf33c6023d73067436110048a3536d

**Comment by ramvikrams** [comment]
yes sir done the changes

File: tests/test_series.py [hunk 8]
```diff
@@ -1237,8 +1237,8 @@ def cond2(x: pd.Series[int]) -> pd.Series[bool]:
     cond3 = pd.Series([False, True, True])
     check(assert_type(s.where(cond3, other=0), "pd.Series[int]"), pd.Series, int)
     
- 
- def test_bitwise_opeator() -> None:
+
```
Comment (Dr-Irv) [inline tests/test_series.py:1240]: have extra spaces on line above.

You need to run `poetry run poe test` and check to see if any errors are reported.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

**Comment by ramvikrams** [comment]
Don't know the error sir works fine on my local system

File: tests/test_series.py (hunk 8 above)
Comment (ramvikrams) [reply tests/test_series.py:1240]: > have extra spaces on line above.
> 
> You need to run `poetry run poe test` and check to see if any errors are reported.

sir on my system no error were reported even after the update

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
you still have issues with the formatting in `test/test_series.py`.  Try running `black tests/test_series.py`.  It should reformat, then commit and push.

If that doesn't work, I'll just have to do the PR, because I can't figure out what you are doing wrong.

**Comment by ramvikrams** [comment]
> 

yes sir

**Comment by ramvikrams** [comment]
done with the test sir and it finished without a error

File: tests/test_series.py [hunk 9]
```diff
@@ -1236,3 +1236,32 @@ def cond2(x: pd.Series[int]) -> pd.Series[bool]:
 
     cond3 = pd.Series([False, True, True])
     check(assert_type(s.where(cond3, other=0), "pd.Series[int]"), pd.Series, int)
+    
+
+def test_bitwise_opeator() -> None:
```
Comment (Dr-Irv) [inline tests/test_series.py:1241]: use `test_bitwise_operator()`  it is misspelled

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
still failing on black

File: tests/test_series.py (hunk 9 above)
Comment (ramvikrams) [reply tests/test_series.py]: done sir

**Comment by ramvikrams** [comment]
> still failing on black

don't know sir on my system it doesn't show any error on black

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
still failing.  Don't know what you are doing wrong.  YOu have extra spaces on line 1239.

**Comment by ramvikrams** [comment]
Don't know sir on my system it shows no error

**Comment by Dr-Irv** [comment]
I still don't know what you are doing wrong, so I fixed it myself.

**Review by Dr-Irv — APPROVED** [review]
thanks @ramvikrams .  Had to finish this myself because I'm spending too much time reviewing failures on your part.  Not sure what you are doing wrong.

**Comment by ramvikrams** [comment]
Thanks sir

**Pull request merged**